tests/test_overscanCorrection.py

```diff
@@ -347,57 +349,177 @@ def test_parallelOverscanCorrection(self):
+        This test adds a large artificial bleed to the overscan region,
+        which should be masked and patched with the median of the
+        other pixels.
         """
-        exposure = self.makeExposure(isTransposed=False)
+        exposure = self.makeExposure(addRamp=True, isTransposed=False)
         detector = exposure.getDetector()
         amp = detector.getAmplifiers()[0]
 
         maskedImage = exposure.getMaskedImage()
-        overscan = afwImage.MaskedImageF(maskedImage, amp.getRawParallelOverscanBBox())
-        overscan.set(400, 0x0, 1)
+        overscanBleedBox = lsst.geom.Box2I(lsst.geom.Point2I(4, 10),
+                                           lsst.geom.Extent2I(2, 3))
+        overscanBleed = afwImage.MaskedImageF(maskedImage, overscanBleedBox)
+        overscanBleed.set(110000, 0x0, 1)
 
         statBefore = computeImageMedianAndStd(exposure.image[amp.getRawDataBBox()])
 
-        config = ipIsr.IsrTask.ConfigClass()
-        config.overscan.doParallelOverscan = True
-        isrTask = ipIsr.IsrTask(config=config)
-        oscanResults = isrTask.overscan.run(exposure, amp)
+        for fitType in ('MEDIAN', 'MEDIAN_PER_ROW', 'POLY'):
+            # We only test these three types as this should cover the
+            # scalar calculations, the generic vector calculations,
+            # and the specific C++ MEDIAN_PER_ROW case.
+            exposureCopy = exposure.clone()
+            config = ipIsr.IsrTask.ConfigClass()
+            config.overscan.doParallelOverscan = True
+            config.overscan.parallelOverscanMaskGrowSize = 1
+            config.overscan.fitType = fitType
+            isrTask = ipIsr.IsrTask(config=config)
 
-        self.assertIsInstance(oscanResults, pipeBase.Struct)
-        self.assertIsInstance(oscanResults.imageFit, float)
-        self.assertIsInstance(oscanResults.overscanFit, float)
-        self.assertIsInstance(oscanResults.overscanImage, afwImage.ExposureF)
+            # This next line is usually run as part of IsrTask:
+            isrTask.overscan.maskParallelOverscan(exposureCopy, detector)
+            oscanResults = isrTask.overscan.run(exposureCopy, amp)
 
-        statAfter = computeImageMedianAndStd(exposure.image[amp.getRawDataBBox()])
-        self.assertLess(statAfter[0], statBefore[0])
+            self.assertIsInstance(oscanResults, pipeBase.Struct)
+            if fitType == 'MEDIAN':
+                self.assertIsInstance(oscanResults.imageFit, float)
+                self.assertIsInstance(oscanResults.overscanFit, float)
+            else:
+                self.assertIsInstance(oscanResults.imageFit, np.ndarray)
+                self.assertIsInstance(oscanResults.overscanFit, np.ndarray)
+            self.assertIsInstance(oscanResults.overscanImage, afwImage.ExposureF)
+
+            statAfter = computeImageMedianAndStd(exposureCopy.image[amp.getRawDataBBox()])
+            self.assertLess(statAfter[0], statBefore[0])
```

Comment:
For this case, with the bleed, the std after correction does not go to zero for the `MEDIAN` or `POLY` case but it is significantly reduced (from 2.87 raw/median to 0.7 (median_per_row) or 0.17 (poly).  Which makes sense given the bleed interruption and the shape of the ramp.   I think that the reduction of the std should be checked here, which shows that the correction is working.